phpstan-baseline.neon

```diff
@@ -0,0 +1,67 @@
+parameters:
```

Comment:
in dieser datei befindet sich die baseline, d.h. alle warnings die wir jetzt erstmal ignorieren wollen

die warnings hier drinnen fallen aktuell in folgende kategorien
- warnings mit ursache in externer SimpleSaml lib
- warnings die kommen, da die abhängigkeiten aus der package.yml (hier konkret yform, yrewrite) nicht bekannt sind, da composer die klassen aus den addons nicht kennt

Reply:
@bloep @gharlan um statische code analyse für addons zu betreiben die zum beispiel auf ycom oder yrewrite aufbauen wäre es gut wenn die klassen aus den addons zur verfügung stehen würden.

bin am überlegen ob man überall ne composer.json reinlegen will und das autoloading etc. alles konfiguriert, oder ob man ggf. sowas wie ein redaxo<->composer plugin schreiben könnte/sollte, dass basierend auf den konventionen die wir im redaxo haben (zumindest im core und den meisten FOR addons) beliebige pakete zur verfügung stellen könnte (und deren klassen).

ob sowas überhaupt geht, weiß ich nicht, aber ich wollts mal reinwerfen

Reply:
ich denke als einfachsten wäre es, die Packages mit composer.json auszustatten (zumal wir ja schon 3 im repo mit einer haben).
wenn das für die Zwecke hier net ausreicht, könnte man immer noch über ein composer plugin nachdenken. :wink:

Reply:
https://github.com/yakamara/redaxo_yform/pull/961